File: 1319-unique-number-of-occurrences/1319-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer, Integer> freqMap = new HashMap<>();
        
        for (int num : arr) {
            freqMap.put(num, freqMap.getOrDefault(num, 0) + 1);
        }
        
        HashSet<Integer> occurances = new HashSet<>();
        
        for(int count : freqMap.values()){
            if(!occurances.add(count))
                return false;
        }
        return true;
    }
}

File: 1319-unique-number-of-occurrences/1319-unique-number-of-occurrences.kt
class Solution {
    fun uniqueOccurrences(arr: IntArray): Boolean {
        val freqMap = mutableMapOf<Int, Int>()

        for(num in arr){
            freqMap.put(num, freqMap.getOrDefault(num, 0) + 1)
        }

        val freqSet = freqMap.values.toSet()

        return arr.toSet().size == freqSet.size
    }
}